mlton/atoms/prim-tycons.fun

```diff
@@ -234,4 +234,64 @@ fun layoutApp (c: t,
       else normal ()
    end
 
+fun layoutFormal (c: t, args: (Layout.t * ({isChar: bool}* BindingStrength.t)) vector) =
+   let
+      local
+         open Layout
+         open BindingStrength
+      in
+         val mayAlign = mayAlign
+         val seq = seq
+         val str = str
+      end
+      val ({isChar}, lay) =
+         case Vector.length args of
+            0 => if equals (c, tuple)
+                    then (({isChar = false}, str "unit"))
+                 else
+                    ({isChar = equals (c, defaultChar ())},
+                     (* always use canonical names, even if synonyms exist *)
+                     if equals (c, real RealSize.R32)
```

Comment:
And, are these special-cases really necessary?  It seems that the `layout c` would yield exactly these strings.